yetanotherco/aligned_layer
# feat: make signing eip 712 compliant

> [!IMPORTANT]  
> Added `chain` param to `submit` and `submit_multiple` in the SDK. This is needed to get the `BatcherPaymentService` contract address.

> [!IMPORTANT]  
> The version of the Aligned contracts was changed from =0.8.12 to ^0.8.12 to enable compatibility with open-zeppelin 5.0.0 (which is needed to use EIP712). We still use open-zeppelin from eigenlayer-middleware for all the imports but just for the EIP712 we use open-zeppelin 5.0.0.

> [!NOTE]  
> The GAP variable was lowered by 1 because of the added NONCED_VERIFICATION_DATA_TYPEHASH bytes32 constant.

> [!NOTE]  
> As stated in the code's comment, VerificationData was used as a bytes32 instead of a VerificationData struct because we don't have the necessary fields in the contract (BatcherPaymentService) to use the VerificationData struct. Also, chain_id is not part of the type hash because it is now part of the domain.

If you want to read more about the [EIP-712](https://eips.ethereum.org/EIPS/eip-712).

# Description

- This PR makes the `NoncedVerificationData` signing EIP712 compliant. The EIP712Domain has:

  ```
  name: Aligned
  version: 1
  chain_id: <current_chain_id>
  verifying_contract: <current_payment_service contract_addr>
  ```

# Deploying
## Devnet:

`make anvil_add_type_hash_to_batcher_payment_service`

## Testnet:

- Stage or prod variables should be set accordingly on the contracts/scripts/.env file

`make upgrade_add_type_hash`

# Working with Metamask

https://github.com/user-attachments/assets/decd274b-6423-4d8e-89e3-718d0f49fa9c

# To Test
- Run `make deps`.
- Run everything as usual and make sure it is working properly.
